source/val/validation_state.h

```diff
@@ -154,7 +154,7 @@ class ValidationState_t {
   bool IsOpcodeInCurrentLayoutSection(SpvOp op);
 
   DiagnosticStream diag(spv_result_t error_code) const;
```

Comment:
My goal is to remove this overload, and then we can remove the int overload I added in this CL and just have the Instruction variant. This removes the cases where we set the line based on where the parser is, vs setting it based on the instruction.